### ISSUE TITLE
No artifacts found matching '*-386.msi' path

### ISSUE THREAD

**Issue by mh-cbon** [appveyor/ci#1752]
in my script

```yml
artifacts:
  - path: '*-386.msi'
    name: msi-x86
  - path: '*-amd64.msi'
    name: msi-x64
```

in my build

```sh
DIR "%APPVEYOR_BUILD_FOLDER%"
 Volume in drive C has no label.
 Volume Serial Number is 1CB8-D4AA
 Directory of c:\gopath\src\github.com\mh-cbon\go-msi
08/28/2017  10:46 AM    <DIR>          .
08/28/2017  10:46 AM    <DIR>          ..
...
08/28/2017  10:46 AM         3,141,632 go-msi-386-1.0.2.msi
08/28/2017  10:46 AM         3,141,632 go-msi-386.msi
08/28/2017  10:46 AM         3,313,664 go-msi-amd64-1.0.2.msi
08/28/2017  10:46 AM         3,313,664 go-msi-amd64.msi
08/28/2017  10:46 AM         3,025,420 go-msi.1.0.2.nupkg
...
              25 File(s)     37,981,411 bytes
              10 Dir(s)  29,231,226,880 bytes free
echo "%APPVEYOR_BUILD_FOLDER%"
"c:\gopath\src\github.com\mh-cbon/go-msi"
Discovering tests...OK
Collecting artifacts...
No artifacts found matching '*-386.msi' path
No artifacts found matching '*-amd64.msi' path
```

It s unclear. Is It ?

**Comment by IlyaFinkelshteyn** [comment]
Please provide the whole build configuration in YAML format.

**Comment by mh-cbon** [comment]
https://github.com/mh-cbon/go-msi/blob/master/appveyor.yml

https://ci.appveyor.com/project/mh-cbon/go-msi

does it help ?

**Comment by IlyaFinkelshteyn** [comment]
Can you please check what is current directly with `pwd` or `cd` command in the end of `before_test`?

**Comment by mh-cbon** [comment]
from previous experiments, it is c:\gopath\src\github.com\mh-cbon/go-msi

Could it be that the commit is not a tag, thus the error message is just totally misleading ?

**Comment by IlyaFinkelshteyn** [comment]
No, message is not totally misleading. Can you please send me a link to the build where you do exactly `cd` or `pwd` in the end of `before_test` commands?

**Comment by IlyaFinkelshteyn** [comment]
Actually please do `cd && dir *msi`

**Comment by mh-cbon** [comment]
https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.171
https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.167


1.0.179 with  your requested addition is running https://ci.appveyor.com/project/mh-cbon/go-msi

**Comment by mh-cbon** [comment]
https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.179#L598

**Comment by IlyaFinkelshteyn** [comment]
It exposed a bug, I filed an issue https://github.com/appveyor/ci/issues/1753. For now workaround is something like `clone_folder: c:\gopath\src\github.com\%APPVEYOR_ACCOUNT_NAME%\%APPVEYOR_PROJECT_SLUG%`

**Comment by mh-cbon** [comment]
thanks. 

________

I previously noted that the value of `APPVEYOR_PROJECT_SLUG` was unexpected (or i misunderstood the doc maybe).

Doc says `APPVEYOR_PROJECT_SLUG - project slug (as seen in project details URL)` but when i tried it i got something like `go-msi` only.

so later i switched to `clone_folder: c:\gopath\src\github.com\%APPVEYOR_REPO_NAME%`

**Comment by mh-cbon** [comment]
Just to let you know, it does happen also on tag commit.

https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.184

___

i dont quiet understand how this could fix `clone_folder: c:\gopath\src\github.com\%APPVEYOR_ACCOUNT_NAME%\%APPVEYOR_PROJECT_SLUG%`

its the same as `clone_folder: c:\gopath\src\github.com\%APPVEYOR_REPO_NAME%`

https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.185

```sh
Build started
git clone -q --branch=master https://github.com/mh-cbon/go-msi.git c:\gopath\src\github.com\mh-cbon/go-msi
git checkout -qf 927525bd57e46a77fe52b386b56d9c162506f342
Running Install scripts
echo "APPVEYOR_PROJECT_SLUG=%APPVEYOR_PROJECT_SLUG%"
"APPVEYOR_PROJECT_SLUG=go-msi"
echo "APPVEYOR_PROJECT_NAME=%APPVEYOR_PROJECT_NAME%"
"APPVEYOR_PROJECT_NAME=go-msi"
echo "APPVEYOR_ACCOUNT_NAME=%APPVEYOR_ACCOUNT_NAME%"
"APPVEYOR_ACCOUNT_NAME=mh-cbon"
set GH_APP=%APPVEYOR_PROJECT_NAME%
set GH_USER=%APPVEYOR_ACCOUNT_NAME%

```

**Comment by mh-cbon** [comment]
Or maybe its a slash issue, is it ?

**Comment by mh-cbon** [comment]
it is https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.187

**Comment by mh-cbon** [comment]
hi, sorry to ring you @IlyaFinkelshteyn  but see that build with a tag, it says it upload, but i can t see the resulting files in my gh release.

So i wonder, where did they go ? In wrong repo ? 

https://ci.appveyor.com/project/mh-cbon/go-msi/build/1.0.196

```
Discovering tests...OK
Collecting artifacts...
Found artifact 'go-msi-386.msi' matching '*-386.msi' path
Found artifact 'go-msi-amd64.msi' matching '*-amd64.msi' path
Uploading artifacts...
[1/2] go-msi-386.msi (3,141,632 bytes)...100%
[2/2] go-msi-amd64.msi (3,313,664 bytes)...100%
never-fail jfrog bt pc --user %GH_USER% --key %BT_KEY% --licenses=MIT --vcs-url=https://github.com/%APPVEYOR_REPO_NAME% %GH_USER%/msi/%GH_APP%
```

 but no file here https://github.com/mh-cbon/go-msi/releases/tag/1.0.2

__________

do you know how long i ll be stuck ? If that shorts i ll wait for your updates and can test for you. If its long, i ll just use my own artillery to upload my artifacts.

**Comment by IlyaFinkelshteyn** [comment]
Sorry I am not familiar with custom commands your are trying to execute. Why do not use [AppVeyor built-in GitHub publishing](https://www.appveyor.com/docs/deployment/github/) BTW?

**Comment by mh-cbon** [comment]
i do, https://github.com/mh-cbon/go-msi/blob/master/appveyor.yml#L81

but since recently it appears it stopped to upload appropriately so i took my own artillery. I d prefer not, its a supplementary step, but i really want to make it work.
See my previous message. (Note since then, i did arrange all the things with my own tool, so you ll find the files, i could not wait any longer)

**Comment by IlyaFinkelshteyn** [comment]
Build log and link you send in [previous post](https://github.com/appveyor/ci/issues/1752#issuecomment-325656186) does not use AppVeyor GitHub provider, it is difficult to guess what was wrong with.

**Comment by mh-cbon** [comment]
sorry, i did not understand 
https://github.com/mh-cbon/go-msi/blob/4a4c69097ba03ba14e30c2ad5a985d95833b6bac/appveyor.yml#L74
